src/core/tsi/ssl_transport_security.cc

```diff
@@ -65,6 +65,7 @@
 
 // --- Constants. ---
 
+#define TSI_SSL_MAX_BIO_WRITE_ATTEMPTS 100
```

Comment:
As we discussed offline, let's do a follow up PR to expose `100` as another parameter.

Reply:
SG, filed and a bug for tracking and CC-ed you.